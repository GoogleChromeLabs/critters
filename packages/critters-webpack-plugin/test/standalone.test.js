/**
 * Copyright 2018 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */

import { compile, compileToHtml, readFile } from './_helpers.js';

function configure(config) {
  config.module.rules.push(
    {
      test: /\.css$/,
      loader: 'css-loader'
    },
    {
      test: /\.html$/,
      loader: 'file-loader?name=[name].[ext]'
    }
  );
}

test('webpack compilation', async () => {
  const info = await compile('fixtures/raw/index.js', configure);
  expect(info.assets).toHaveLength(2);
  expect(await readFile('fixtures/raw/dist/index.html')).toMatchSnapshot();
});

describe('Usage without html-webpack-plugin', () => {
  let output;
  beforeAll(async () => {
    output = await compileToHtml('raw', configure);
  });

  it('should process the first html asset', () => {
    const { html, document } = output;
    expect(document.querySelectorAll('style')).toHaveLength(1);
    expect(document.getElementById('unused')).toBeNull();
    expect(document.getElementById('used')).not.toBeNull();
    expect(document.getElementById('used').textContent).toMatchSnapshot();
    expect(html).toMatchSnapshot();
  });
});
