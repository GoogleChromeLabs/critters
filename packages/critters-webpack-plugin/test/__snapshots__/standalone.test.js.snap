// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Usage without html-webpack-plugin should process the first html asset 1`] = `
"h1 {
  color: green;
}"
`;

exports[`Usage without html-webpack-plugin should process the first html asset 2`] = `
"<!DOCTYPE html><html><head>
    <title>Basic Demo</title>
    
    <style id=\\"used\\">h1 {
  color: green;
}</style>
  </head>
  <body>
    <h1>Some HTML Here</h1>
  

</body></html>"
`;

exports[`webpack compilation 1`] = `
"<!DOCTYPE html><html><head>
    <title>Basic Demo</title>
    <style>body {
  padding-left: 11em;
  font-family: \\"Times New Roman\\", times, serif;
  color: purple;
  background-color: #d8da3d;
}

ul.navbar {
  list-style-type: none;
  padding: 0;
  margin: 0;
  position: absolute;
  top: 2em;
  left: 1em;
  width: 9em;
}

h1 {
  font-family: helvetica, arial, sans-serif;
}

ul.navbar li {
  background: white;
  margin: 0.5em 0;
  padding: 0.3em;
  border-right: 1em solid black;
}

ul.navbar a {
  text-decoration: none;
}

ul.navbar:not(.hidden) li:hover a {
  color: black;
}

ul.navbar:not(:hover) li:hover a {
  color: black;
}

ul.navbar li:hover a {
  color: red;
}

a:link {
  color: blue;
}

a:visited {
  color: purple;
}

footer {
  margin-top: 1em;
  padding-top: 1em;
  border-top: thin dotted;
}

.clazz\\\\:colon {
  color: green;
}</style>
  </head>
  <body>
    <ul class=\\"navbar clazz:colon\\">
      <li>
        <a href=\\"index.html\\">Home page</a>
      </li>
      <li>
        <a href=\\"musings.html\\">Musings</a>
      </li>
    </ul>
    <h1>My first styled page</h1>
    <p>Welcome to my styled page!</p>
    <footer>Made 5 April 2004</footer>
  <script type=\\"text/javascript\\" src=\\"bundle.js\\"></script>

</body></html>"
`;
